src/wrapped_radix_engine_toolkit/low_level/transaction_intent.rs

```diff
@@ -36,26 +39,52 @@ impl TransactionIntent {
     }
 
     pub fn compile(&self) -> BagOfBytes {
-        let scrypto_intent: ScryptoIntent = self.clone().into();
-        let compiled = RET_intent_compile(&scrypto_intent)
-            .expect("Should always be able to compile an Intent");
-
-        compiled.into()
+        compile_intent(ScryptoIntent::from(self.clone()))
+            .expect("Should always be able to compile an Intent")
     }
 }
 
 impl From<TransactionIntent> for ScryptoIntent {
     fn from(value: TransactionIntent) -> Self {
-        Self {
-            header: value.header.into(),
-            instructions: ScryptoInstructions(
-                value.manifest.instructions().clone(),
-            ),
-            blobs: value.manifest.blobs().clone().into(),
-            message: value.message.into(),
-        }
+        into_scrypto(&value.header, &value.manifest, &value.message)
     }
 }
+
+fn into_scrypto(
```

Comment:
cannot use `From<TransactionIntent>` inside the constructor of `TransactionIntent`... duh... so we use this global fn, which is also used by our `impl From<TransactionIntent> for ScryptoIntent` implementation